io-sim/src/Control/Monad/IOSim/Types.hs

```diff
@@ -145,9 +145,12 @@ data SimA s a where
   SetWallTime  ::  UTCTime -> SimA s b  -> SimA s b
   UnshareClock :: SimA s b -> SimA s b
 
-  NewTimeout   :: DiffTime -> (Timeout (IOSim s) -> SimA s b) -> SimA s b
-  UpdateTimeout:: Timeout (IOSim s) -> DiffTime -> SimA s b -> SimA s b
-  CancelTimeout:: Timeout (IOSim s) -> SimA s b -> SimA s b
+  NewTimeout       :: DiffTime -> (Timeout (IOSim s) -> SimA s b) -> SimA s b
+  NewRegisterDelay :: DiffTime -> (TVar s Bool -> SimA s b) -> SimA s b
+  NewThreadDelay   :: DiffTime -> SimA s b -> SimA s b
+
+  UpdateTimeout :: Timeout (IOSim s) -> DiffTime -> SimA s b -> SimA s b
+  CancelTimeout :: Timeout (IOSim s) -> SimA s b -> SimA s b
```

Comment:
Just as a grouping thing: I'd keep the `NewTimeout`, `UpdateTimeout`, `CancelTimeout` together in one group, since they're one API together.

The `NewRegisterDelay` and `NewThreadDelay` are each independent timer/timeout/delay features and APIs.

Reply:
Done